index.d.ts

```diff
@@ -1,31 +1,64 @@
+import { Input, InputObject } from 'into-stream'
+import { Stream } from 'stream'
+import { BrotliOptions, ZlibOptions } from 'zlib'
+
+type EncodingToken = 'br' | 'deflate' | 'gzip' | 'identity';
+
+export interface FastifyCompressOptions {
+  brotliOptions?: BrotliOptions;
+  customTypes?: RegExp;
+  encodings?: EncodingToken[];
+  forceRequestEncoding?: EncodingToken;
+  global?: boolean;
+  inflateIfDeflated?: boolean;
+  onInvalidRequestPayload?: (encoding: string, request: FastifyRequest, error: Error) => Error | undefined | null;
+  onUnsupportedEncoding?: (encoding: string, request: FastifyRequest, reply: FastifyReply) => string | Buffer | Stream;
+  onUnsupportedRequestEncoding?: (encoding: string, request: FastifyRequest, reply: FastifyReply) => Error | undefined | null;
+  requestEncodings?: EncodingToken[];
+  threshold?: number;
+  zlib?: unknown;
+  zlibOptions?: ZlibOptions;
+}
+
+interface RouteCompressOptions extends Pick<FastifyCompressOptions,
+  | 'brotliOptions'
+  | 'customTypes'
+  | 'encodings'
+  | 'inflateIfDeflated'
+  | 'onUnsupportedEncoding'
+  | 'threshold'
+  | 'zlib'
+  | 'zlibOptions'
+> {}
+
+interface RouteDecompressOptions extends Pick<FastifyCompressOptions,
+  | 'forceRequestEncoding'
+  | 'onInvalidRequestPayload'
+  | 'onUnsupportedRequestEncoding'
+  | 'requestEncodings'
+  | 'zlib'
+> {}
+
+interface FastifyCompressRouteOptions {
+  compress?: false | RouteCompressOptions;
+  decompress?: false | RouteDecompressOptions;
+}
+
+export interface RouteOptions extends fastify.RouteOptions, FastifyCompressRouteOptions {}
+
+declare module 'fastify' {
+  export interface RouteShorthandOptions<
+    RawServer extends RawServerBase = RawServerDefault
+  > extends FastifyCompressRouteOptions { }
 
-declare module "fastify" {
   interface FastifyReply {
     compress(input: Stream | Input | InputObject): void;
   }
-}
-
-type EncodingToken = 'br' | 'deflate' | 'gzip' | 'identity'
 
-export interface FastifyCompressOptions {
-  global?: boolean
-  threshold?: number
-  customTypes?: RegExp
-  zlib?: NodeModule
-  brotliOptions?: BrotliOptions
-  zlibOptions?: ZlibOptions
-  inflateIfDeflated?: boolean
-  onUnsupportedEncoding?: (encoding: string, request: FastifyRequest, reply: FastifyReply) => string | Buffer | Stream
-  encodings?: Array<EncodingToken>
-  requestEncodings?: Array<EncodingToken> 
-  forceRequestEncoding?: EncodingToken
-  onUnsupportedRequestEncoding?: (encoding: string, request: FastifyRequest, reply: FastifyReply) => Error | undefined | null
-  onInvalidRequestPayload?: (encoding: string, request: FastifyRequest, error: Error) => Error | undefined | null
+  export interface RouteOptions extends FastifyCompressRouteOptions {}
 }
 
-declare const fastifyCompress: FastifyPlugin<FastifyCompressOptions>
-export default fastifyCompress;
+declare const fastifyCompress: FastifyPluginCallback<FastifyCompressOptions>
```

Comment:
`FastifyPlugin` is deprecated. Changed to the recommended `FastifyPluginCallback`.
This is a breaking change.